### PR TITLE
chore(script): Script migrate_node.sh supports using specified config file when using shell config

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 
https://github.com/apache/incubator-pegasus/issues/1368

- Manual test
image

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1660,6 +1663,11 @@ function run_migrate_node()
                 CLUSTER="$2"
                 shift
                 ;;
+            -p|--config)
+                CONFIG="$2"
+                CONFIG_SPECIFIED=1
```
Comment (acelyc111) [inline run.sh:1668]: `CONFIG_SPECIFIED` can be removed, just judge whether `CONFIG` is empty or not?

File: run.sh [hunk 2]
```diff
@@ -1703,14 +1711,22 @@ function run_migrate_node()
...
     echo "APP=$APP"
     echo "TYPE=$TYPE"
     echo
     cd ${ROOT}
     echo "------------------------------"
-    ./scripts/migrate_node.sh $CLUSTER $NODE "$APP" $TYPE
+    if [ ${CONFIG_SPECIFIED} -eq 0 ]; then
+        ./scripts/migrate_node.sh $CLUSTER $NODE "$APP" $TYPE
+    else
+       ./scripts/migrate_node.sh $CLUSTER $NODE "$APP" $TYPE $CONFIG_SPECIFIED
```
Comment (acelyc111) [inline run.sh:1728]:
```suggestion
        ./scripts/migrate_node.sh $CLUSTER $NODE "$APP" $TYPE $CONFIG_SPECIFIED
```

Comment (acelyc111) [inline run.sh:1728]: Should use `$CONFIG` instead of `$CLUSTER`?

File: scripts/migrate_node.sh [hunk 3]
```diff
@@ -20,37 +20,52 @@ set -e
...
+  node=$2
+  app_name=$3
+  type=$4
+else
+  CONFIG_SPECIFIED=1
+  config=$1
+  node=$2
+  app_name=$3
+  type=$4
+fi
```
Comment (acelyc111) [inline scripts/migrate_node.sh:48]:
```suggestion
if [ $# -eq 4 ]; then
  CONFIG_SPECIFIED=0
  cluster=$1
else
  CONFIG_SPECIFIED=1
  config=$1
fi
node=$2
app_name=$3
type=$4
```

**Comment by empiredan** [comment]
If it's necessary to be compatible with the previous format of arguments for `migrate_node.sh`, we could add an option `-t` for `migrate_node.sh`:

- `migrate_node.sh -t config <arguments ...>` means the first argument is config, or
- `migrate_node.sh -t cluster <arguments ...>` or just `migrate_node.sh <arguments ...>` (`-t` is omitted) means the first argument is cluster.

**Review by empiredan — APPROVED** [review]

File: scripts/migrate_node.sh [hunk 4]
```diff
@@ -20,37 +20,58 @@ set -e
...
 fi
 
 echo "UID=$UID"
 echo "PID=$PID"
 echo
 
-echo "set_meta_level steady" | ./run.sh shell --cluster $cluster &>/tmp/$UID.$PID.pegasus.set_meta_level
-
-echo ls | ./run.sh shell --cluster $cluster &>/tmp/$UID.$PID.pegasus.ls
+if [ ${CONFIG_SPECIFIED} -eq 0 ]; then
```
Comment (acelyc111) [inline scripts/migrate_node.sh:68]: `CONFIG_SPECIFIED` can be replaced by `cluster`?

**Review by acelyc111 — APPROVED** [review]

**Review by empiredan — APPROVED** [review]

**Pull request merged**